rollup-common.js

```diff
@@ -123,7 +119,42 @@ Object.entries(stableProperties).forEach(([prop, mangle], i) => {
   }
 });
 
-const generateTerserOptions = (nameCache = null) => ({
+/**
+ * Prefixes all class properties with the given prefix character. This is to
+ * effectively namespace private properties on subclassable objects to avoid
+ * accidental collisions when users (or our own packages) subclass them. We
+ * choose a different prefix character per compilation unit (package), which
+ * guarantees that automatically chosen mangled names don't collide between our
+ * own packages.
+ *
+ * Note that Terser has no understanding of classes; class properties getting
+ * prefix treatment are identified via convention, where class properties are
+ * authored with double `__` whereas normal object properties that can be
+ * mangled un-prefixed use single `_`.
+ *
+ * Prefix characters chosen in the "Other Letter (Lo)" unicode category
+ * (https://codepoints.net/search?gc=Lo) are valid in JS identifiers, should be
+ * sufficiently collision-proof to hand-authored code, and are unlikely to be
+ * chosen (at least default) by minifiers.
+ */
+const addedClassPrefix = new WeakSet();
+const prefixClassProperties = (context, nameCache, prefix) => {
+  // Only prefix class properties once per options context, as a perf optimization
+  if (nameCache && !addedClassPrefix.has(context)) {
+    const {
+      props: {props},
+    } = nameCache;
+    for (const p in props) {
+      if (p.startsWith('$__') && !props[p].startsWith(prefix)) {
```

Comment:
Why is it `$__` and not `__` and why do you need to check if the prop doesn't start with the prefix given the optimization?

Reply:
Al had a comment in there about this:
https://github.com/Polymer/lit-html/blob/15624b3b163643a8979151fffaaee7d9ee988d14/rollup-common.js#L218-L219

Checking if the prop starts with the prefix makes it safe to run `prefixClassProperties` more than once on the same `nameCache`; I was just thinking that since the code could get refactored, we might not catch if we were double-prefixing properties.

Reply:
Added comments.